aries_cloudagent/protocols/out_of_band/v1_0/manager.py

```diff
@@ -200,30 +197,36 @@ async def create_invitation(
                     "Cannot create public invitation with no public DID"
                 )
 
-            if metadata:
-                raise OutOfBandManagerError(
-                    "Cannot store metadata on public invitations"
-                )
-
-            invi_msg = InvitationMessage(
+            invi_msg = InvitationMessage(  # create invitation message
                 label=my_label or self._session.settings.get("default_label"),
                 handshake_protocols=handshake_protocols,
-                request_attach=message_attachments,
```

Comment:
According to RFC 0434, an existing connection can only be reused based on a public DID. If we cannot create a public invitation with attachments then it will not support the last 2 cases in the following table:
https://github.com/hyperledger/aries-rfcs/tree/master/features/0434-outofband#responses